core/src/main/java/org/ldaptive/pool/AbstractConnectionPool.java

```diff
@@ -988,7 +988,14 @@ public PooledConnectionStatistics getPooledConnectionStatistics()
     @Override
     public boolean equals(final Object o)
     {
-      return LdapUtils.areEqual(this, o);
+      if (o == this) {
```

Comment:
I'm not sure equality should be defined for component classes. What's the use case?